src/dhcpmon/src/dhcp_mon.c

```diff
@@ -131,7 +134,20 @@ static void timeout_callback(evutil_socket_t fd, short event, void *arg)
         check_dhcp_relay_health(&state_data[i]);
     }
 
-    dhcp_devman_update_snapshot(NULL);
+    /* if dhcp status is indeterminate, do not update the snapshot.
+     * There is a possibility that client dhcp discover packets are
+     * perfectly timed to the boundary of dhcpmon buckets such that
+     * client packet arrives before dhcpmon timer kicks any and
+     * relayed packets go out after dhcpmon timer is serviced. In
+     * this situation dhcp status will be unhealthy, however subsequent
+     * indeterminate state results in consuming those relayed packet
+     * into the snapshot without updating the dhcp status to healthy.
+     * The idea here is to delay snapshoting those packet and compare them
+     * with next opportunity when dhcp packet are received.
+     */
+    if (state_data[0].dhcp_status != DHCP_MON_STATUS_INDETERMINATE) {
```

Comment:
Is it enough to check `state_data[0].dhcp_status != DHCP_MON_STATUS_INDETERMINATE` here? Should you also check that `state_data[1].dhcp_status != DHCP_MON_STATUS_INDETERMINATE`?